db/db_utils.go

```diff
@@ -388,16 +389,23 @@ func Aggregate(collection string, pipeline interface{}, out interface{}) error {
 	// socket timeout, which isn't really an option here. (other
 	// operations had a 90s timeout, which is no longer specified)
 
-	pipe := db.C(collection).Pipe(pipeline)
+	pipe := ConstructPipe(db, collection, pipeline)
+	return unmarshalPipelineResults(pipe, out)
+}
+
+func ConstructPipe(db db.Database, collection string, pipeline interface{}) db.Aggregation {
```

Comment:
I don't know that I agree with this being a helper function, since it's just one line and it's easily readable, and is only used in one place